### ISSUE TITLE
Keep track of recipe's changes

### ISSUE THREAD

**Issue by kelson42** [openzim/zimfarm#498]
For the moment, we don't really keep track of changes on recipes (or this is not visible for end-users). This has two major consequences:
* The transparency about who does what is not secured
* Considering that this is a tool which is a bit technical, there is no way to learn from the changes that other users made.

It would be great to have a kind of revision log (like on Mediawiki).

Keeping track of each revision in json format and having a diff display between revisions would be really helpful.

I'm in favour as well to keep a big more information about each revision like:
* Comment about the revision
* Author of the revision
* Date

**Comment by rgaudin** [comment]
Would the comment be mandatory? I'd advise not to.

**Comment by kelson42** [comment]
> Would the comment be mandatory? I'd advise not to.

Yes, but might help to explain why things have been done. Not always obvious.

**Comment by rgaudin** [comment]
OK, can't wait to see your ticket asking to make it not-mandatory 😀

**Comment by kelson42** [comment]
A system working in a similar manner like the Gandi DNS zone revision system would be fine.

**Comment by kelson42** [comment]
I think this ticket start to be really super urgent because:
* We are more and more people editing in the Zimfarm
* Rules are complext and learning curve is long
* We need more visibility on who does what to be able to better control/fix/learn

We need IMO this very same feature on the CMS.

In term of architecture I would:
* Each time a modification is made we should save in addition the username + comment (mandatory)
* Save full JSON (compressed) of new version of each recipe in a dedicated table
* Display a new tab "History" on recipes allowing to make a diff
* We need a global "recent change" system (a feed on it would be even better.

**Comment by benoit74** [comment]
The idea would be to have a new `schedule_history` table with following columns:

- `id` (primary key)
- `schedule_id`
- `author`
- `date`
- `comment` (optional)
- `config`

The `config` will store schedule configuration at given `date` in JSON format. It means that for "live" config, this config will be available both in `schedule` table (used lived for all tasks related stuff) and in `schedule_history`. This seems to be a good compromise to simplify history tracking / comparison (we will only compare records from the `schedule_history` table).

The UI will display list of history for a given schedule and allow to diff any two of them.

The UI will show JSON-diff computed on-the-fly.
